src/index.js

```diff
@@ -0,0 +1,13 @@
+import { searchCategory } from "./searchBar";
+import { displayCategoryLink } from "./displayUI";
+import { btnClear } from "./displayUI";
+import { singletonInstance } from './mainClass';
+
+singletonInstance.allCategories = searchCategory;
+singletonInstance.linksUpdate = displayCategoryLink;
```

Comment:
as improvement creat set methods for this callback, is nicer to keep encapsulation in class (read more OOP and encapsulation)